detector/aws_instance_invalid_ami.go

```diff
@@ -7,19 +7,30 @@ import (
 	"github.com/wata727/tflint/issue"
 )
 
-func (d *Detector) DetectAwsInstanceInvalidAmi(issues *[]*issue.Issue) {
+type AwsInstanceInvalidAMIDetector struct {
+	*Detector
+}
+
+func (d *Detector) CreateAwsInstanceInvalidAMIDetector() *AwsInstanceInvalidAMIDetector {
+	return &AwsInstanceInvalidAMIDetector{d}
+}
+
+func (d *AwsInstanceInvalidAMIDetector) Detect(issues *[]*issue.Issue) {
 	if !d.Config.DeepCheck {
 		d.Logger.Info("skip this rule.")
 		return
 	}
 
 	validAmi := map[string]bool{}
-	resp, err := d.AwsClient.Ec2.DescribeImages(&ec2.DescribeImagesInput{})
-	if err != nil {
-		d.Logger.Error(err)
-		d.Error = true
+	if d.ResponseCache.DescribeImagesOutput == nil {
+		resp, err := d.AwsClient.Ec2.DescribeImages(&ec2.DescribeImagesInput{})
+		if err != nil {
+			d.Logger.Error(err)
+			d.Error = true
+		}
+		d.ResponseCache.DescribeImagesOutput = resp
```

Comment:
__[gometalinter]__
* [gotype] cannot use resp (variable of type *github.com/aws/aws-sdk-go/service/ec2.DescribeImagesOutput) as *invalid type value in assignment

<!-- Comment by [SideCI](https://sideci.com) -->